app/controllers/courses_controller.rb

```diff
@@ -17,6 +17,11 @@ def create
     end
   end
 
+  def show
+    course = Course.find_by(courseCode: params['id'])
+    render json: {course: course}
+  end
+
```

Comment:
This is a good implementation but I suggest we work with id instead of course code.

Reply:
@czarjulius thanks for that.
I have changed it to ID